src/ov_utils.cpp

```diff
@@ -82,39 +82,48 @@ std::optional<ov::Layout> getLayoutFromRTMap(const ov::RTMap& rtMap) {
     return std::nullopt;
 }
 
-Status validatePluginConfiguration(const plugin_config_t& pluginConfig, const std::string& targetDevice, const ov::Core& ieCore) {
-    auto availableDevices = ieCore.get_available_devices();
-    auto availablePlugins = availableDevices;
+void insertSupportedKeys(std::vector<std::string>& pluginSupportedConfigKeys, const std::string& pluginName, const ov::Core& ieCore) {
     const std::string supportedConfigKey = METRIC_KEY(SUPPORTED_CONFIG_KEYS);
-    std::vector<std::string> allSupportedConfigKeys;
-    for (const auto& plugin : availablePlugins) {
-        std::vector<std::string> pluginSupportedConfigKeys;
-        try {
-            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Validating plugin: {}; configuration", plugin);
-            std::vector<std::string> pluginSupportedConfigKeys2 = ieCore.get_property(plugin, supportedConfigKey).as<std::vector<std::string>>();
-            pluginSupportedConfigKeys = std::move(pluginSupportedConfigKeys2);
-        } catch (std::exception& e) {
-            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from IE when requesting plugin: {}; key: {}; value. Error: {}", plugin, supportedConfigKey, e.what());
-        } catch (...) {
-            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from IE when requesting plugin: {}; key: {}; value.", plugin, supportedConfigKey);
-        }
+    try {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Validating plugin: {}; configuration", pluginName);
+        std::vector<std::string> pluginSupportedConfigKeys2 = ieCore.get_property(pluginName, supportedConfigKey).as<std::vector<std::string>>();
+        pluginSupportedConfigKeys.insert(pluginSupportedConfigKeys.end(), pluginSupportedConfigKeys2.begin(), pluginSupportedConfigKeys2.end());
+    } catch (std::exception& e) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from IE when requesting plugin: {}; key: {}; value. Error: {}", pluginName, supportedConfigKey, e.what());
+    } catch (...) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from IE when requesting plugin: {}; key: {}; value.", pluginName, supportedConfigKey);
```

Comment:
Shouldn't those two be warn/error? What if we fail getting supported keys? Should we just ignore this rather than not load the model?